pkg/datastore/resourceindexer.go

```diff
@@ -52,41 +52,44 @@ func newResourceIndexer(ctx context.Context, logger *zap.Logger, db *gorm.DB) (r
 	return qb, err
 }
 
-//newFieldColumns creates the map from a slice of fieldColumn
-func newFieldColumns(fields []fieldColumn) fieldColumns {
-	result := make(map[string]fieldColumn)
-	for _, field := range fields {
-		result[field.FieldName] = field
-	}
-	return result
-}
-
 //an explicit field means that the column will be named after the field
-func (f fieldColumns) addExplicitFields(names ...string) {
+func (f fieldColumns) addExplicitFields(group string, names ...string) {
 	for _, name := range names {
-		f[name] = fieldColumn{
-			ColumnName: name,
-			FieldName:  name,
-		}
+		f.addFieldColumn(group, fieldName(group, name), name)
 	}
 }
 
 //a dynamic field would have a generated column name
-func (f fieldColumns) addDynamicFields(names ...string) bool {
+func (f fieldColumns) addDynamicFields(group string, names ...string) bool {
 	newField := false
 	for _, name := range names {
-		if _, found := f[name]; !found {
-			columnName := f.newColumnName()
-			f[name] = fieldColumn{
-				ColumnName: columnName,
-				FieldName:  name,
-			}
+		fieldName := fieldName(group, name)
+		if _, found := f[fieldName]; !found {
+			f.addFieldColumn(group, fieldName, f.newColumnName())
 			newField = true
 		}
 	}
 	return newField
 }
 
+func (f fieldColumns) addFieldColumn(group, fieldName, columnName string) {
```

Comment:
```suggestion
func (f fieldColumns) addFieldColumn(fieldName, columnName string) {
```

nitpick: `group` is passed to this function but is unused.

Reply:
good catch, thanks